packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,131 @@
+   * to handle searching. This handler will be called once with an empty {@link SearchQuery.searchTerm}
+   * when search is beginning. 
+   * @param onClosedHandler - This handler will be called when the user finishes searching. Should be
+   * used to return your application to its default, non-search state. The value of {@link SearchQuery.searchTerm}
+   * will be whatever the last query was before ending search.
+   * @param onExecuteHandler - This optional handler will be called whenever the user 'executes' the
+   * search (by pressing enter for example). The value of {@link SearchQuery.searchTerm} is the current 
+   * term the user is searching for. Should be used if your app wants to treat executing searches differently than responding
+   * to changes to the search query.
+   *
+   * @example
+   * ``` ts
+   * search.registerHandlers(
+      query => {
+        console.log(`Update your application to render a change to the search query: ${query.searchTerm}`);
+      },
+      () => {
+        console.log('Update your application to handle the search experience being closed');
+      },
+      query => {
+        console.log(`Update your application to render an executed search result: ${query.searchTerm}`);
+      },
+     );
+   * ```
+   */
+  export function registerHandlers(
+    onChangeHandler: SearchQueryHandler,
+    onClosedHandler: SearchQueryHandler,
+    onExecuteHandler?: SearchQueryHandler,
+  ): void {
+    // TODO: figure out what frame contexts you want to support this in
+    // This is just a guess that I made and should be something you make
+    // an explicit decision about.
+    ensureInitialized(
+      FrameContexts.content,
+      FrameContexts.task,
+      FrameContexts.sidePanel,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    );
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    registerHandler(onChangeHandlerName, onChangeHandler);
+    registerHandler(onClosedHandlerName, onClosedHandler);
+    if (onExecuteHandler) {
+      registerHandler(onExecutedHandlerName, onExecuteHandler);
+    }
+  }
+
+  /**
+   * Allows the caller to unregister for all events fired by the host search experience. Calling
+   * this function will cause your app to stop appearing in the set of search scopes in the host.s
+   */
+  export function unregisterHandlers(): void {
+    // TODO: figure out what frame contexts you want to support this in
+    // This is just a guess that I made and should be something you make
+    // an explicit decision about.
+    ensureInitialized(
+      FrameContexts.content,
+      FrameContexts.task,
+      FrameContexts.sidePanel,
+      FrameContexts.stage,
+      FrameContexts.meetingStage,
+    );
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    // This should let the host know to stop making the app scope show up in the search experience
```

Comment:
Does that mean the user will have no way to 'remove' that default 'app scope' from search UI once the app calls `registerHandlers`? And if there is a way for them 'escape' that default scope, will there also be a way to get back to it somehow?

Reply:
According to the spec, the user can change the search scope and this would only affect whatever the "default" current scope is while in the application